### PR TITLE
Add opcode 0x46 blockreward

### PR DESCRIPTION
**Pull request by realcodywburns**
From #698 and #669 

Simple Summary

This EIP adds an additional opcode to the EVM which will return a finalized blocks reward value. 

See commits and files

### CODE REVIEW

**Comment by nicksavers** [comment]
Comments in issue

@5chdn https://github.com/ethereum/EIPs/issues/698#issuecomment-325140306
> Doesn't this even make sense to include it in EIP-649 #669 ?

@realcodywburns https://github.com/ethereum/EIPs/issues/698#issuecomment-325141216
> It could be useful independent of 649. 649 seems more specifically focused on the mechanics of reward reduction on the mainnet ethereum surrounding the metro forks, whereas the opcode itself could be used universally: testnets and sister chains(etc, music, expanse, rootstock)

@realcodywburns https://github.com/ethereum/EIPs/issues/698#issuecomment-325148831
> For further clarity, `BLOCKREWARD` should return the total value of the Reward Applied(11.3) for block `n` as defined in section 11, Block Finalisation

@axic https://github.com/ethereum/EIPs/issues/698#issuecomment-325156613
> .... it could be useful to have a way to get both the basic and the total reward, perhaps as a single parameter to the opcode (enum or boolean?).

@vbuterin https://github.com/ethereum/EIPs/issues/698#issuecomment-332765558
> I don't think this is worth the effort at the moment given the impending PoS switch. Once the exact long term consensus incentivization structure is finalized, I think it would make more sense to revisit something like this.

@realcodywburns https://github.com/ethereum/EIPs/issues/698#issuecomment-333273922
> The PoS switch has been "impending" for quite some time and seems to be delayed a bit longer. This eip is also applicable to consortium chains, private chains, and others using multiple consensus algorithms.

File: EIPS/eip-opcodeBLKREWARD.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
+## Preamble
+    
+	EIP: <to be assigned>
```
Comment (pirapira) [inline EIPS/eip-opcodeBLKREWARD.md]: Please use 698 and move this file to EIPS/eip-698.md.

File: EIPS/eip-698.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+In the EVM, the 0x40 opcodes are reserved for `Block Information`. Currently reserved opcodes are:
+* `0X40 BLOCKHASH`
+* `0X41 COINBASE`
+* `0X42 TIMESTAMP`
+* `0X43 NUMBER`
+* `0X44 DIFFICULTY`
+* `0X45 GASLIMIT`
+
+This EIP would add an additional opcode, `0x46 BLOCKREWARD`, which would return the block reward for any finalized block. The finalized block reward would include the base reward, uncle payments, and gas.
```
Comment (axic) [inline EIPS/eip-698.md]: I don't think it is possible to get the "final" block reward for the current block since total gas usage cannot be known in advance.

File: EIPS/eip-698.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## Simple Summary
+
+This EIP adds an additional opcode to the EVM which will return a finalized blocks reward value. 
+
+
+
+## Abstract
+
+In the EVM, the 0x40 opcodes are reserved for `Block Information`. Currently reserved opcodes are:
```
Comment (axic) [inline EIPS/eip-698.md]: Is this list relevant to the current EIP?

File: EIPS/eip-698.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+## Specification
+
+
+After block `n` all clients should process opcode `0x46` as follows:  
+
+* Value: `0x46`
+* Mnemonic: `BLOCKREWARD`
+* δ:` 0` nothing removed from stack
+* α:`1` block reward added to stack
```
Comment (axic) [inline EIPS/eip-698.md]: Since the total reward cannot be known for the current block, it may be more useful to:
- have an input parameter (block number)
- return 4 outputs: base reward, uncle reward, variable (gas) reward (as proposed here https://github.com/ethereum/EIPs/issues/698#issuecomment-325156613) - maybe even outputting total gas consumed

It may be possible to implement this change similar to how #210 is done, that would make it clear that all this detail can only be retrieved for past blocks.

**Comment by Arachnid** [comment]
This is a courtesy notice to let you know that the format for EIPs has been modified slightly. If you want your draft merged, you will need to make some small changes to how your EIP is formatted:

 - Frontmatter is now contained between lines with only a triple dash ('---')
 - Headers in the frontmatter are now lowercase.

If your PR is editing an existing EIP rather than creating a new one, this has already been done for you, and you need only rebase your PR.

In addition, a continuous build has been setup, which will check your PR against the rules for EIP formatting automatically once you update your PR. This build ensures all required headers are present, as well as performing a number of other checks.

Please rebase your PR against the latest master, and edit your PR to use the above format for frontmatter. For convenience, here's a sample header you can copy and adapt:

```
---
eip: <num>
title: <title>
author: <author>
type: [Standards Track|Informational|Meta]
category: [Core|Networking|Interface|ERC] (for type: Standards Track only)
status: Draft
created: <date>
---
```

File: EIPS/eip-698.md [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+* `0X43 NUMBER`
+* `0X44 DIFFICULTY`
+* `0X45 GASLIMIT`
+
+This EIP would add an additional opcode, `0x46 BLOCKREWARD`, which would return the block reward for any finalized block. The finalized block reward would include the base reward, uncle payments, and gas.   
+
+## Motivation
+
+
+Per EIP-649 ( #669 ) periodic block reward reductions/variance are now planned in the roadmap, however, this EIP is consensus system agnostic and is most useful in decentralized pool operations and for any contract that benefits from knowing a block reward payout(i.e. Merge mined tokens)
```
Comment (nicksavers) [inline EIPS/eip-698.md]: [EIP-649](https://eips.ethereum.org/EIPS/eip-649) is now Final. You may want to update the text to reflect this.
